### PR TITLE
pl-order-blocks - Adding support for horizontal order blocks problems

### PR DESCRIPTION
**Pull request by CoffeePoweredComputers**
This pull request adds an attribute `"block-orientation"` to the pl-order-blocks element which can be set to `"horizontal"` when `solution-placement="bottom"` in order to achieve the following problem format:

micro-parsons-demo

This feature was inspired by ["Regex Parsons: Using Horizontal Parsons Problems to Scaffold Learning Regex"](https://doi.org/10.1145/3488042.3489968)](https://dl.acm.org/doi/10.1145/3488042.3489968) and Zihan Wu's work on Micro Parsons problems (@amy21206)

### CODE REVIEW

**Comment by mwest1066** [comment]
This is super exciting!

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 1]
```diff
@@ -45,11 +45,17 @@ class FormatType(Enum):
     CODE = "code"
 
 
+class BlockOrientationType(Enum):
+    DEFAULT = "default"
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:49]: I would prefer that the default actually be called "vertical" not just "default"

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 2]
```diff
@@ -481,6 +496,20 @@ def render(element_html: str, data: pl.QuestionData) -> str:
             "Drag answer tiles into the answer area to the " + dropzone_layout + ". "
         )
 
+        if block_orientation is BlockOrientationType.HORIZONTAL:
+            if dropzone_layout != "bottom":
+                raise Exception(
+                    'The block-orientation attribute may only be "horizontal" when solution-placement is "bottom".'
+                )
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:503]: Is this restriction really necessary? I don't really care if a person wants to do it this way. If their problem only consists of just a few small blocks it would probably work just fine.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 3]
```diff
@@ -481,6 +496,20 @@ def render(element_html: str, data: pl.QuestionData) -> str:
...
+            if check_indentation:
+                raise Exception(
+                    'The indentation attribute may not be used when block-orientation is "horizontal".'
+                )
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:507]: Thanks for adding this check, this one we do want 👍

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 4]
```diff
@@ -481,6 +496,20 @@ def render(element_html: str, data: pl.QuestionData) -> str:
...
+            if max_indent != MAX_INDENTION_DEFAULT:
+                raise Exception(
+                    'The max-indent attribute may not be used when block-orientation is "horizontal".'
+                )
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:511]: I think this check is unnecessary because it is redundant--we already disallow having max-indent unless the indentation is on, which we are already disallowing with the check right before this.

File: exampleCourse/questions/element/orderBlocks/question.html [hunk 5]
```diff
@@ -135,3 +135,31 @@
...
+    <strong>Example 6:</strong> Create a SQL command that selects all students who are enrolled in computer science.
+  </p>
+
+  <p>
+    This example uses the attributes <code>selection-placement="bottom"</code>
+    and <code>block_orientation="horizontal"</code> to create horizontal arrangement 
+    of the source blocks.
+  </p>
+  <br>
+  <b>Note:</b> This model of block arrangement, particularly for the context of CS, is attributed to
```
Comment (SethPoulsen) [inline exampleCourse/questions/element/orderBlocks/question.html:150]: "CS" should be "computer science"

File: exampleCourse/questions/element/orderBlocks/question.html [hunk 6]
```diff
@@ -135,3 +135,31 @@
...
+  <pl-answer correct="true">SELECT</pl-answer>
+  <pl-answer correct="true">students</pl-answer>
+  <pl-answer correct="true">FROM</pl-answer>
+  <pl-answer correct="true">enrollments</pl-answer>
+  <pl-answer correct="true">WHERE</pl-answer>
+  <pl-answer correct="true">major</pl-answer>
+  <pl-answer correct="true">=</pl-answer>
+  <pl-answer correct="true">"computer science"</pl-answer>
+</pl-order-blocks>
+
```
Comment (SethPoulsen) [inline exampleCourse/questions/element/orderBlocks/question.html:165]: very cool example!

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 7]
```diff
@@ -156,6 +162,7 @@ def prepare(element_html: str, data: pl.QuestionData) -> None:
         "format",
         "code-language",
         "allow-blank",
+        "block-orientation",
     ]
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:166]: When this feature existed previously it was done with `inline=true/false`, but I actually don't like that much.

I'm not sure `block-orientation` is the right thing to call it either though. It doesn't actually change the orientation of the blocks per se, but the way that they are layed out with respect to one another. What would people think about `block-flow=vertical/horizontal`. @mwest1066 @nwalters512 @jonatanschroeder @armandofox  do you have thoughts about what this attribute should be called?

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 8]
```diff
@@ -465,13 +477,16 @@ def render(element_html: str, data: pl.QuestionData) -> str:
 
         for option in student_previous_submission:
             submission_indent = option.get("indent", None)
+
             if submission_indent is not None:
+                if block_orientation is BlockOrientationType.HORIZONTAL:
+                    raise Exception(
+                        'When block-orientation is "horizontal" indentaion is not supported. Please remove "indent" attributes from all pl-answer blocks.'
+                    )
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:485]: I don't think this is needed, don't we already have this case covered?

Having an `indent` attribute on a block is already disallowed unless indentation=true, but you are already disallowing that below.

File: exampleCourse/questions/element/orderBlocks/question.html [hunk 9]
```diff
@@ -135,3 +135,31 @@
...
+    and <code>block_orientation="horizontal"</code> to create horizontal arrangement 
+    of the source blocks.
+  </p>
+  <br>
+  <b>Note:</b> This model of block arrangement, particularly for the context of CS, is attributed to
+  <a href="https://doi.org/10.1145/3488042.3489968">"Regex Parsons: Using Horizontal Parsons Problems to Scaffold Learning Regex"</a>
+
+</pl-question-panel>
+
+<pl-order-blocks answers-name="sql-query" grading-method="ranking" format="code" code-language="sql" solution-placement="bottom" block-orientation="horizontal">
```
Comment (SethPoulsen) [inline exampleCourse/questions/element/orderBlocks/question.html:155]: If you are not specifying a `ranking` attribute on each line, then you should be using `grading-method=ordered`. Actually, we should be throwing an error for this but we are not :(  (fixing that is outside the scope of this PR though)

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css [hunk 10]
```diff
@@ -70,12 +73,26 @@ li.ui-state-highlight {
...
 
+.pl-block-horizontal > .list-group{
+  overflow-x: scroll;
+}
+
+.pl-block-horizontal >  ul > li,
+.pl-block-horizontal > .list-group {
+  display: inline-block !important;
+  float: none !important;
+  white-space: nowrap;
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:84]: I think you actually want `display: inline` and `float: left` here. That solves the issues you were having with placeholders showing in the wrong place in some cases.

Also, get rid of the `!important`s

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js [hunk 11]
```diff
@@ -152,6 +152,12 @@ window.PLOrderBlocks = function (uuid, options) {
       let leftDiff = calculateIndent(ui, placeholder.parent());
       placeholder[0].style.marginLeft = leftDiff + 'px';
       placeholder[0].style.height = ui.item[0].style.height;
+
+      //enable width if the dropzone isn't bottom
+      if (options.dropzoneLayout === 'pl-order-blocks-bottom'){
+        placeholder[0].style.width = ui.item[0].style.width;
+      }
+
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js:160]: This shouldn't be necessary once you fix the styles as I said above.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css [hunk 12]
```diff
@@ -10,6 +10,7 @@ div.pl-order-blocks-bottom {
   /* This particular style renders the dropzones in a single column vertically */
   width: 100%;
   margin: 8px;
+  min-height: 130px;
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:13]: why is this here? kind of a weird thing to hardcode, I would like to avoid that

**Review by SethPoulsen — CHANGES_REQUESTED** [review]
Looks great overall! Just lots of minor issues

**Comment by SethPoulsen** [comment]
Having equations in the blocks can lead to blocks of different heights: 
image

We definitely want this fixed because I know people want to use this for math.

**Comment by SethPoulsen** [comment]
Your build is failing on javascript linting. Make sure you are running the proper linters & formatters (see docs: https://prairielearn.readthedocs.io/en/latest/dev-guide/#coding-style)

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 10 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:84]: this does make it so that if the blocks don't fit on one line, they go to the next line instead of making it scroll. If you want it to scroll, you should tweak `overflow` in some way, I think. 

It might be nice in the future to have it configurable between scrolling, or flowing on to the next line. But don't worry about that for now, just set it up the way you like it and we can make it configurable later if people want it.

File: exampleCourse/questions/element/orderBlocks/question.html (hunk 6 above)
Comment (jonatanschroeder) [reply exampleCourse/questions/element/orderBlocks/question.html:165]: the only issue here is that `WHERE "computer science" = major` is also a valid answer, but this is not a dealbreaker...

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 7 above)
Comment (jonatanschroeder) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:166]: I actually prefer inline instead of vertical/horizontal. This should allow future improvements where wrapping is allowed (instead of scrolling) for long content.

**Comment by CoffeePoweredComputers** [comment]
> Having equations in the blocks can lead to blocks of different heights: image
> 
> We definitely want this fixed because I know people want to use this for math.

By fix, should all blocks take on the height of the tallest block or did you have a different behavior in mind?

**Comment by SethPoulsen** [comment]
> By fix, should all blocks take on the height of the tallest block or did you have a different behavior in mind?

hmm, actually thinking about it again, I don't think we necessarily want to force blocks to be ceratain sizes. Just go ahead and leave it as-is for now.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 7 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:166]: I think I agree with @jonatanschroeder as well. Id initially had it as a boolean attribute but switched to try and be consistent with selection-placement being a string attribute.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 8 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:485]: Ah! Didn't catch that. It is now removed.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 12 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:13]: With minwidth: 
image


Without min-width:
image

When I was testing, dragging between boxes that were resizing and and starting out with a 0-width construction area felt pretty annoying. I wanted it to feel more like the traditional order-blocks where the height of the columns is set based on the tallest column. The main thing I'd like to avoid is resizing while the user is dragging so perhaps dropzone height should be set as the problem's tallest block?

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js (hunk 11 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js:160]: I think this is still needed since it's setting the placeholder width. When removed and with the previously mentioned css changes in place I get this: 
image

File: exampleCourse/questions/element/orderBlocks/question.html (hunk 9 above)
Comment (CoffeePoweredComputers) [reply exampleCourse/questions/element/orderBlocks/question.html:155]: Fixed. Might open another PR for this since you mentioned it.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 7 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:166]: You can go ahead and use `inline=true/false` then. I don't have strong feelings about it.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 1 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:49]: disregard this if you end up just using `inline=true/false` you can just delete this enum altogether.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js (hunk 11 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js:160]: Oh, sorry, I got confused about what this was doing. Shouldn't we be doing this based on if `inline=true`, not based on where the dropzone is?

You can pass the `inline` boolean into the `PLOrderBlocks` creation function call as a property on the `options` object. Then this would just be `if (options.inline) {....`

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 12 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:13]: oh I see. Ya it is nice to have it start with some height. 

I don't think resizing during dragging is necessarily a problem though, that's what the behavior is now. A few options are starting same height as smallest or biggest block, or just starting as same size as the starting zone. I'm not sure what's best without playing with it a bit. 

We can always just ship something and then tweak it later if we feel like.

File: exampleCourse/questions/element/orderBlocks/question.html (hunk 5 above)
Comment (CoffeePoweredComputers) [reply exampleCourse/questions/element/orderBlocks/question.html:150]: Fixed.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 4 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:507]: Removed.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 2 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:499]: Ah, fair. I didn't consider the case where there might be just a few small blocks. Removed.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 10 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css]: Swapping over to `display: block` and `float: left` did the trick. The debug on the browser was saying due to the use of float it was automatically switching to `display: block` . That said, I can't quite figure out how to get scrolling working with this configuration.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 7 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:166]: Sounds good! I'll go with that.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 12 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:13]: When stacking vertically I don't think its as much of a problem. However it felt a bit clunky to me when dragging things around horizontally and having it dynamically resize vertically. I'll go with setting it to the biggest block for right now and see what you think.

Comment (jonatanschroeder) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:13]: A size with `rem` units may be a better option here.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 10 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css]: Couldn't get scrolling working when using `display: inline` but I did finally find an attribute that seems to have fixed the alignment issues.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 13]
```diff
@@ -437,7 +438,10 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     element = lxml.html.fragment_fromstring(element_html)
     answer_name = pl.get_string_attrib(element, "answers-name")
     format = pl.get_enum_attrib(element, "format", FormatType, FormatType.DEFAULT)
-
+    inline = pl.get_boolean_attrib(element, "inline", INLINE_DEFAULT)
+    dropzone_layout = pl.get_string_attrib(
```
Comment (eliotwrobson) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:442]: It seems like this variable is only supposed to take on a finite set of values? In that case, should use `get_enum_attrib` instead of `get_string_attrib` with an appropriately defined enum.

Also, be sure to update the docs.

Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:442]: This attribute predates this PR so I didn't want to mess with it initially. @SethPoulsen is there any reason it was a string attribute or should I go ahead and switch to an `enum`?

Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:442]: dropzone layout can definitely be an enum, it just hasn't been converted yet. (The enum system is new).

Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:442]: Cool, updated. :)

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 12 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css]: updated!

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 14]
```diff
@@ -544,6 +562,10 @@ def render(element_html: str, data: pl.QuestionData) -> str:
             "allow_feedback_badges": not all(
                 block.get("badge_type", "") == "" for block in student_submission
             ),
+            "block_layout": "pl-block-horizontal" if inline else "",
```
Comment (eliotwrobson) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:565]: Can the values for the keys `"block_layout"` and `"dropzone_layout"` be read from a hardcoded dictionary (instead of if statements like now)? The mapping logic seems to be the same for both, so reading the values from a dict will clean up the new code.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css [hunk 15]
```diff
@@ -43,6 +44,7 @@ li.ui-state-highlight {
   list-style-type: none;
   background-color: #90a4ae !important;
   min-width: 3em !important;
+  margin-bottom: 5px !important;
```
Comment (SethPoulsen) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:47]: do we need the `!important` here? its kind of hack and should be avoided whenever possible.

**Review by SethPoulsen — APPROVED** [review]
Excellent work @CoffeePoweredComputers. I'm excited for people to start using this.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css (hunk 15 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css]: Ah, yeah that needs to go and is now removed. Sorry, Copilot has a tendency to autocomplete those in and I sometime forget to remove them. 😞

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 14 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:565]: Personally I don't see a need for this. The if statements are quite simple as it is. Factoring it out to a dictionary means you have to go look somewhere else in the code to see what its doing. 

I don't have particularly strong feelings for or against though, I'm cool with whatever.

Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:565]: I might also lean towards keeping it as is. Though I'd agree Python's ternaries are less than pretty at times I do think its nice to just have the values front and center.

Comment (eliotwrobson) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:565]: It's less about reducing the amount of logic and more about reducing the number of string literals in the code. If changed, it's true you can't see the values directly on the same line, but you're relying on having copied the same strings correctly in multiple places.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py [hunk 16]
```diff
@@ -437,7 +442,13 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     element = lxml.html.fragment_fromstring(element_html)
     answer_name = pl.get_string_attrib(element, "answers-name")
     format = pl.get_enum_attrib(element, "format", FormatType, FormatType.DEFAULT)
-
+    inline = pl.get_boolean_attrib(element, "inline", INLINE_DEFAULT)
+    dropzone_layout = pl.get_enum_attrib(
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: Is there a reason this isn't named `solution_placement` instead of `dropzone_layout`?

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css
Comment (nwalters512) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:1]: Why were the permission bits changed on this and `pl0order-blocks.js`?

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css [hunk 17]
```diff
@@ -70,12 +72,29 @@ li.ui-state-highlight {
   margin-bottom: 5px;
 }
 
+.pl-block-horizontal > .list-group {
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:75]: Why do these class names use a `pl-block-` prefix instead of the usual `pl-order-blocks-`?

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.mustache [hunk 18]
```diff
@@ -5,15 +5,19 @@ $(function() {
 
     new window.PLOrderBlocks(UUID, {
         maxIndent: {{max_indent}},
-        enableIndentation: {{check_indentation}}
+        inline: {{inline}},
+        enableIndentation: {{check_indentation}},
+        dropzoneLayout: "{{dropzone_layout}}",
+        blockLayout: "{{block_layout}}",
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.mustache:11]: I can't figure out where these are actually read inside the `PLOrderBlocks` class. If they aren't in fact needed, please remove them.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 16 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: Not sure. That variable name is used elsewhere and predates this PR so I was just continuing to use it to be consistent.

Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: Not sure. That variable name was used elsewhere in the code so I just kept using it to be consistent. Happy to change it if you prefer the former.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.mustache (hunk 18 above)
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.mustache:11]: Removed.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css
Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.css:1]: Not actually sure how that happened... Regardless, I reverted them.

File: apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py (hunk 16 above)
Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: Ya, this variable name is from like 4 or 5 years ago. I don't think anyone here has any particular attachment to it. But We should have consistency across the whole file of course.

Comment (SethPoulsen) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: I'm not really bothered at the moment because the naming is consistent, I'm ok leaving it. If @nwalters512 thinks 'solution_placement' is a much better variable name, then when we change it we should change it everywhere so we still have consistency.

Comment (nwalters512) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: I'd encourage you to change this in a fast-follow PR. Even though the variable name is consistent across functions, it's completely inconsistent with the attribute name, which I'd like to remedy.

Comment (CoffeePoweredComputers) [reply apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py:446]: @nwalters512 Agreed and will do! I'll hammer that out this weekend.
